xwiki-platform-core/xwiki-platform-ckeditor/xwiki-platform-ckeditor-plugins/src/main/webjar/xwiki-link/plugin.js

```diff
@@ -522,13 +522,22 @@
     return {
       id: 'optionsToggle',
       type: 'html',
-      html: '<div class="linkOptionsToggle">' +
+      html: '<button class="linkOptionsToggle" type="button">' +
               '<label class="cke_dialog_ui_labeled_label">' +
                 '<span class="arrow arrow-right"></span> ' +
                 CKEDITOR.tools.htmlEncode(editor.localization.get('xwiki-link.options')) +
               '</label>' +
-            '</div>',
+            '</button>',
       onLoad: function() {
+        // Since we do not (and cannot without deeper changes) use the 'button' type, 
+        // we need to add this element explicitely to the Dialog focus list.
+        // We need to hardcode the position since we do not have access to the setupFocus function to reorder the list
+        // relative to native tab order.
+        // The four elements before this button are: display link, page selection *3
+        this.getDialog().addFocusable(this.getElement() , 4);
+        // Without this, the keyboard press on this focusable element will trigger the click twice...
+        this.getElement().removeAllListeners();
```

Comment:
Wouldn't it be safer to remove only the ``keydown`` listener that seems to be responsible for the double click? See https://github.com/ckeditor/ckeditor4/blob/82796d03d0ebbe34d48f3b65c6dde27107812a51/plugins/dialog/plugin.js#L709-L713 .

Reply:
hmm, indeed there is no simple way to remove only the ``keydown`` listener (without having a reference to the listener) https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_dom_element.html#method-removeListener . So I'm fine to keep your current solution to remove all listeners.

Reply:
Note that there shouldn't be any other way to interact with this dropdown button other than a click (mouse or keyboard). I doubt this'd break any other feature.